AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/MutationEvent+Query.swift

```diff
@@ -6,17 +6,57 @@
 //
 
 import Amplify
+import Dispatch
 
 extension MutationEvent {
     static func pendingMutationEvents(forModelId modelId: Model.Identifier,
                                       storageAdapter: StorageEngineAdapter,
                                       completion: DataStoreCallback<[MutationEvent]>) {
+
+        pendingMutationEvents(forModelIds: [modelId], storageAdapter: storageAdapter, completion: completion)
+    }
+
+    static func pendingMutationEvents(forModelIds modelIds: [Model.Identifier],
+                                      storageAdapter: StorageEngineAdapter,
+                                      completion: DataStoreCallback<[MutationEvent]>) {
         let fields = MutationEvent.keys
```

Comment:
As discussed, let's move `maxNumberOfPredicates` into a constant on `StorageAdapter` so we can reuse across our code. We'll also explore making this asynchronous, or moving to the `run` idiom we use above, so we're not using multiple ways of doing synchronous database calls in our code.

As above, **_after_** we get this working, let's benchmark whether an `IN` query would performs better than explicitly declaring multiple `=` conditions. (Again, it may not since I think the `IN` query basically translates to multiple equals behind the scenes.)

Reply:
> in regards to the async pattern, what do you mean by the run idiom above?

Above, rather than using a `DispatchSemaphore`, we drop down to the underlying `run` method that synchronously executes arbitrary SQL. My suggestion is that we pick a pattern to do synchronous data access, and use that pattern everywhere, rather than jumping around between patterns.

That said, there may be good reasons to keep them separate: maybe the place we use `run` above requires custom SQL, and isn't supported via the adaptor. Maybe the things we're trying to express in the code where we use `DispatchSemaphore` is clearer by using the StorageAdaptor methods, which aren't synchronous. So this is just a suggestion that we explore it. If it makes the code more understandable to have different patterns, keep them (but maybe document at the call site why we depart from the standard pattern)